### ISSUE TITLE
User Lat/Lon Sometimes Show all Zeros (user entered locations in FTM 2017)

### ISSUE THREAD

**Issue by paulp575** [ShammyLevva/FTAnalyzer#41]
For SOME of my user entered locations in FTM 2017, the lat/lon showing in FTAnalyzer is 0.0000000/0.0000000 with an entry in the Geocode Status column as GEDCOM/User. However FTM shows at lat/lon.

1. Enter location in Place field in FTM.
2. FTM responds with a question mark showing the location is not in the Place Name Authority.
3. Open Google Search and enter name of location.
4. Look to see if the Google search results shows a Wikipedia page for the location (normally does).
5. Open the Wikipedia page.
6. Copy lat/lon on Wikipedia page.
7. Paste lat/lon from Wikipedia page into the Location field in the Places Workspace in FTM.
8. Verify the pushpin is in the correct location; if not. manually move the pushpin to the proper location.
9. FTM2017 accepts the location.
10. Export file as a GEDCOM.
11. Open FTAnalyzer and import GEDCOM.
12. Click on Maps - Run Google Geocoder to Find Locations.
13. View Locations Geocoding Status Report; sort by latitude.

I have verified that ALL my locations in FTM 2017 have a lat/long.

I have attached an abbreviated GEDCOM file showing an individual with a location of Deansville, Dane County, Wisconsin.

Desktop:
 - OS: Windows 7 Pro (64-bit) w/32 GB RAM
 - Browser: Firefox v 61.0.1

[Paul Yelk 2018-03-18 - 2018-10-18 - Zimprich.zip](https://github.com/ShammyLevva/FTAnalyzer/files/2493551/Paul.Yelk.2018-03-18.-.2018-10-18.-.Zimprich.zip)

**Comment by ShammyLevva** [comment]
This looks like it's to do with intermediate locations. When It loads a location such as Deansville, Dane County, Wisconsin, USA. It creates 4 locations namely:

Deansville, Dane County, Wisconsin, USA
Dane County, Wisconsin, USA
Wisconsin, USA
USA

This is so the things like the tree view of the locations works, and you can click on USA to get everyone in USA or click on Wisconsin, USA to get everyone in the state of Wisconsin. 

It would appear that the only locations showing as unknown are the ones that were created as intermediaries.

**Comment by paulp575** [comment]
On 10/19/2018 12:18 PM, Alexander Bisset wrote:
>
> This looks like it's to do with intermediate locations. When It loads 
> a location such as Deansville, Dane County, Wisconsin, USA. It creates 
> 4 locations namely:
>
> Deansville, Dane County, Wisconsin, USA
> Dane County, Wisconsin, USA
> Wisconsin, USA
> USA
>
> This is so the things like the tree view of the locations works, and 
> you can click on USA to get everyone in USA or click on Wisconsin, USA 
> to get everyone in the state of Wisconsin.
>
> It would appear that the only locations showing as unknown are the 
> ones that were created as intermediaries.
>

I am totally confused by your e-mail.

Intermediaries?

What create the 4 intermediaries? FTAnalyzer?

Why should that make a difference because sometimes it's unknown if the 
location is a city/township or a county. I do have some entries that 
give "(something), Wisconsin". And with many counties across the USA 
having the same name as cities/towns within their county, unless it's 
specifically spelled out in the source document, there's no way of 
knowing. When I list the places in FTM using the non-hierarchical view, 
sometimes there's entries for say "Green County, Wisconsin" (without a 
lower jurisdiction). Other times the source document may give a city/towns.

Paul Yelk

**Comment by ShammyLevva** [comment]
The full location is loaded from your GEDCOM. In order to build a tree of locations FTAnalyzer splits the locations and if they don’t exist adds each of the next level up in turn. 

So if your file has place, county, state, country (it will assume country if it doesn’t exist). Then it loads the full place, county, state, country plus any lat/long that exists. 

In order to show this at multiple levels it also checks for county, state, country and state, country and just country. That’s what I called intermediate locations. 

So if you haven’t specifically got a location for Wisconsin, USA then FTAnalyzer will create one. This created location won’t have a default lat long as there wasn’t one in the file. Geocoding the location will fix this. 

You can view the locations tree on the locations tab.

**Comment by paulp575** [comment]
On 10/19/2018 5:11 PM, Alexander Bisset wrote:
>
> The full location is loaded from your GEDCOM. In order to build a tree 
> of locations FTAnalyzer splits the locations and if they don’t exist 
> adds each of the next level up in turn.
>
> So if your file has place, county, state, country (it will assume 
> country if it doesn’t exist). Then it loads the full place, county, 
> state, country plus any lat/long that exists.
>
> In order to show this at multiple levels it also checks for county, 
> state, country and state, country and just country. That’s what I 
> called intermediate locations.
>
> So if you haven’t specifically got a location for Wisconsin, USA then 
> FTAnalyzer will create one. This created location won’t have a default 
> lat long as there wasn’t one in the file. Geocoding the location will 
> fix this.
>
> You can view the locations tree on the locations tab.
>

I just took a look at the Places tab in FTM and used the hierarchical view.

There are people linked to the following locations which also show lat/long:

USA (actually United States)
Wisconsin
Dane County

So appears there is still something amiss here.

Paul Yelk

**Comment by ShammyLevva** [comment]
Hi Paul,

I'm replying directly on the github website rather than via email so the whole thread is here. Hence not quoting your email.

Did you see the picture I'd attached? The program has created the locations such as Wisconsin as you didn't have anyone in that sample file that had a location that was ONLY Wisconsin. So nowhere in the GEDCOM file was there a location that was only Wisconsin they were always more accurate than that.

Since in order to create a hierarchy, so it can list everyone in that state, it has to extract the states as an individual location. You didn't have it as a specific location so that's why it shows as not searched and lat/long of 0,0. 

NB. I'm only viewing your sample test file not your whole file. Can you check your sample file and confirm what it is you are seeing with that sample that isn't in the picture I sent.

**Comment by paulp575** [comment]
1. I'm not finding the picture you say you attached.

2. You are correct that the sample GEDCOM file I sent you does not have a location of Wisconsin; Dane County; Wisconsin; or United States so I have attached my complete GEDCOM file. I opened it in Notepad++ and did a search for only:

USA -  8 hits
Wisconsin - 28 hits
Dane County - 36 hits

If I'm understanding your information, then by using the complete GEDCOM file, there should be no places (USA; Wisconsin; or Dane County) with lat/lon of 0.0000000/0.0000000.

[Paul Yelk 2018-03-18 - 2018-10-18.zip](https://github.com/ShammyLevva/FTAnalyzer/files/2498681/Paul.Yelk.2018-03-18.-.2018-10-18.zip)

Paul Yelk

**Comment by ShammyLevva** [comment]
I posted the Pic on Facebook Messenger not here sorry. Here it is 
![image](https://user-images.githubusercontent.com/29687786/47262055-715f6580-d4d5-11e8-9a94-d2253ec7f863.png)

**Comment by ShammyLevva** [comment]
Just loaded your larger file and can see no places with GEDCOM/User Data as the tag and 0/0. All of the 0/0s are tagged as Not Searched. So I must be missing something?

**Comment by ShammyLevva** [comment]
Right I've found the bug. 

The issue was those created locations. When it first found a location if the county, state or country location didn't exist it added it with a 0,0. The issue was when it later found, for example the state details in the file, it thought oh I've already created that I don't need to create it again and so it was ignoring your lat long in the file. I've applied a fix for this. Thanks for the report.

I'm in process of building a new v7.0.0.0 so it might be a day or so before this is available.
